bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/DataFlowDiagramConverter.java

```diff
@@ -0,0 +1,444 @@
+                }
+            }
+        }
+        return builder.toString().trim();
+    }
+
+    private void parseBehavior(Node node, Pin outpin, String lines, DataFlowDiagram dfd, DataDictionary dd) {
+        String[] behaviorStrings = lines.split("\n");
+        var behavior = node.getBehaviour();
+        for (String behaviorString : behaviorStrings) {
+            if (behaviorString.contains("forward ")) {
+                String packet = behaviorString.split(" ")[1];
+
+                Pin inpin = dfd.getFlows().stream().filter(flow -> flow.getDestinationNode() == node)
+                        .filter(flow -> flow.getEntityName().equals(packet)).map(Flow::getDestinationPin).findAny().orElse(null);
+
+                var assignment = ddFactory.createForwardingAssignment();
+                assignment.setOutputPin(outpin);
+                assignment.getInputPins().add(inpin);
+                behavior.getAssignment().add(assignment);
+            } else if (behaviorString.contains("set ")) {
+                String regex = "\\bset\\s+(\\S+)\\s+=\\s+(.+)\\b";
+
+                Pattern pattern = Pattern.compile(regex);
+                Matcher matcher = pattern.matcher(behaviorString);
+                if (!matcher.find() || matcher.groupCount() != 2) {
+                    logger.error("Invalid behavior string:");
+                    logger.error(behaviorStrings);
+                    continue;
+                }
+                String variable = matcher.group(1);
+                String typeName = variable.split("\\.")[0];
+                String valueName = variable.split("\\.")[1];
+
+                Label value = dd.getLabelTypes().stream().filter(labelType -> labelType.getEntityName().equals(typeName))
+                        .flatMap(labelType -> labelType.getLabel().stream()).filter(label -> label.getEntityName().equals(valueName)).findAny()
+                        .orElse(null);
+
+                Assignment assignment = ddFactory.createAssignment();
+
+                assignment.getInputPins().addAll(behavior.getInPin());
+                assignment.setOutputPin(outpin);
+                assignment.getOutputLabels().add(value);
+
+                behavior.getAssignment().add(assignment);
+
+                Term term = behaviorConverter.stringToTerm(matcher.group(2));
+
+                assignment.setTerm(term);
+            }
+        }
+    }
+
+    private void putValue(Map<Node, Map<Pin, String>> nestedHashMap, Node node, Pin pin, String value) {
+        nestedHashMap.computeIfAbsent(node, k -> new HashMap<>()).put(pin, value);
+    }
+
+    /**
+     * Converts a Web Editor DFD format file into a DataFlowDiagramAndDictionary object.
+     * @param inputFile The path of the input file in Web Editor DFD format.
+     * @return DataFlowDiagramAndDictionary object representing the converted data flow diagram and dictionary.
+     */
+    public DataFlowDiagramAndDictionary webToDfd(String inputFile) {
```

Comment:
Please have a look at the `.datadictionary` of the manually modeled online shop DFD. I think you can resolve more names in the same way instead of having the default empty `aName`. This should also be tested in a test case.

Reply:
Where do you encounter `aName` while converting webdfd <-> dfd?

Reply:
I have test case ensuring the names of all nodes remain the same for every conversion

Reply:
Oh sorry this was inprecise I meant the names of behaviors and pins in the dictionary, they could also be derived from the other names, for an example see the manually created dictionary in the online shop dfd example